morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -88,13 +139,73 @@ instance
       MorphoStdHeader {..} = morphoHeaderStd
 
 instance MPretty MorphoBlockTx where
-  mpretty = mpretty . morphoBlockGenTx
+  mpretty (MorphoBlockTx genTx genTxId) =
+    "MorphoBlockTx(transaction:"
+      <+> mpretty genTx
+      <> ", transactionId:"
+      <+> mpretty genTxId
+      <> ")"
+
+instance MPretty (ChainHash (MorphoBlock h c)) where
+  mpretty GenesisHash = "GenesisHash"
+  mpretty (BlockHash h) = "BlockHash(" <> mpretty h <> ")"
+
+instance MPretty (HeaderHash a) => MPretty (Tip a) where
+  mpretty TipGenesis = "TipGenesis"
+  mpretty (Tip s h b) =
+    "Tip(slot:"
+      <+> mpretty s
+      <> ", hash:"
+      <+> mpretty h
+      <> ", blockNo:"
+      <+> mpretty b
+      <> ")"
 
 instance MPretty Tx where
   mpretty (Tx vote) = mpretty vote
 
 instance MPretty Vote where
-  mpretty = viaShow
+  mpretty (Vote block signature) =
+    let bytes = powBlockRefToBytes block
+        pubKey = recoverPublicKey signature bytes
+     in "Vote(block:"
+          <+> mpretty block
+          <> ", byPubkey:"
+          <+> mpretty pubKey
+          <> ")"
+
+instance MPretty PowBlockNo where
+  mpretty (PowBlockNo n) = pretty n
+
+instance MPretty PowBlockHash where
+  mpretty (PowBlockHash bytes) = pretty $ Text.take 8 (bytesToHex bytes)
+
+instance MPretty PowBlockRef where
+  mpretty (PowBlockRef number hash) =
+    "PowBlock(number:" <+> mpretty number <> ", hash:" <+> mpretty hash <> ")"
+
+instance MPretty Checkpoint where
+  mpretty (Checkpoint powBlockRef signatures) =
+    let bytes = powBlockRefToBytes powBlockRef
+        pubKeys = map (`recoverPublicKey` bytes) signatures
+     in "Checkpoint(block:"
+          <+> mpretty powBlockRef
+          <> ", signedByPubkeys:"
+          <+> mpretty pubKeys
+          <> ")"
```

Comment:
Currently this is of type `[Maybe PublicKey]`, which then prints as

> [Just(PublicKey("3ea6ebf3")), Just(PublicKey("ec33a368"))]

We can make this a bit nicer with

```haskell
instance MPretty Checkpoint where
  mpretty (Checkpoint powBlockRef signatures) =
    let bytes = powBlockRefToBytes powBlockRef
        showSignature sig = maybe "<unknown>" mpretty (recoverPublicKey sig bytes)
        pubKeys = map showSignature signatures
     in "Checkpoint(block:"
          <+> mpretty powBlockRef
          <> ", signedByPubkeys:"
          <+> list pubKeys
          <> ")"
```